jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -535,6 +536,21 @@ static InputStream asInputStream(Request request)
         return Content.Source.asInputStream(request);
     }
 
+    /**
+     * Get a known {@link MimeTypes.Type} from the request {@link HttpHeader#CONTENT_TYPE}, if any.
+     * @param request The request.
+     * @return A {@link MimeTypes} or {@code null} if the {@code Content-Type} is not set or not known.
+     */
+    static MimeTypes.Type getContentMimeType(Request request)
+    {
+        HttpField contentType = request.getHeaders().getField(HttpHeader.CONTENT_TYPE);
+        if (contentType instanceof MimeTypes.ContentTypeField contentTypeField)
+            return contentTypeField.getMimeType();
+        if (contentType == null)
+            return null;
+        return MimeTypes.CACHE.get(contentType.getValue());
```

Comment:
If it is a custom content-type, are we really returning `null`?
If that's the intent, should not this method be called `getKnownMimeType(Request)` or similar?

Reply:
I thought it was clear enough that the name was `getContentMimeType` rather than `getContentType`.      But if it is confusing, then I'll just delete the method, as it is only a convenience method